packages/corelib/src/playout/infinites.ts

```diff
@@ -481,7 +504,12 @@ export function getPieceInstancesForPart(
 		// }
 	}
 
-	return result
+	const uniqueIds = new Set(result.map((piece) => piece._id))
+	return result.filter((piece) => {
+		const available = uniqueIds.has(piece._id)
+		uniqueIds.delete(piece._id)
+		return available
+	})
```

Comment:
A filter should be a pure expression. Deleting entries from a set outside of the scope violates that. The use case seems more like reducing a list than filtering.
To keep the function pure, you could build the unique ids list while iterating.
```typescript
return result.reduce(({ pieces, uniqueIds }, piece) => {
  if (!uniqueIds.has(piece._id)) {
    pieces.push(piece)
    uniqueIds.add(piece._id)
  }
 return { pieces, uniqueIds }
}, { pieces: [], uniqueIds: new Set() }).pieces
```

Or you could use the accumulator to check if a piece with an equivalent id has already been added. 
```typescript
return result.reduce((pieces, piece) => {
  if (pieces.some({ _id }) => _id === piece._id)) {
    return pieces
  }
  return [...pieces, piece]
}, [])
```

The first example runs in `O(n*lg(n))` while the second example runs in`O(n*n)`. I don't think that we will encounter enough data in this function, for the time complexity to matter.

Reply:
Changed to using second suggestion.